src/io.jl

```diff
@@ -0,0 +1,145 @@
+"""
+    readsxmodel(file; raw_dict=false, ST=ConstrainedLinearControlContinuousSystem, N=Float64, kwargs...)
+
+Read a SX model file.
+
+### Input
+
+- `file`      -- the filename of the SX file (in XML format)
+- `raw_dict`  -- (optional, default: `false`) if `true`, return the raw dictionary with
+                 the objects that define the model (see Output below), without
+                 actually computing the `HybridSystem`; otherwise, instantiate a
+                 `HybridSystem` with the given assumptions
+- `ST`        -- (optional, default: `nothing`) assumption
+                 for the type of mathematical system for each mode
+- `N`         -- (optional, default: `Float64`) numeric type of the system's coefficients
+
+### Output
+
+Hybrid system that corresponds to the given SX model and the given assumptions
+on the system type. If `raw_dict=true`; otherwise, a dictionary with the Julia
+expression objects that define the model. The keys of this dictionary are:
+
+- `automaton`
+- `variables`
+- `transitionlabels`
+- `invariants`
+- `flows`
+- `resetmaps`
+- `switchings`
```

Comment:
What is the difference between `resetmaps` and `switchings`? Is there a reason to have both?

Reply:
the association that i did (naively, subject to discussion) is:

- `resetmaps` field in HybridSystems <->  assignments in SX
- `switchings` field in HybridSystems <-> guards in SX

Reply:
That seems perfect :)